hplsql/src/main/java/org/apache/hive/hplsql/udf/Udf.java

```diff
@@ -66,48 +64,23 @@ public ObjectInspector initialize(ObjectInspector[] arguments) throws UDFArgumen
     if (!(arguments[0] instanceof StringObjectInspector)) {
       throw new UDFArgumentException("First argument must be a string");
     }
-    SessionState sessionState = SessionState.get();
-    if (sessionState != null) {
-      // we are still in HiveServer, get the source of the HplSQL function and store it.
-      functionDefinition = loadSource(sessionState, functionName(arguments[0]));
-    }
     queryOI = (StringObjectInspector)arguments[0];
+    funcDefOI = (StringObjectInspector)arguments[arguments.length-1];
     argumentsOI = arguments;
     return PrimitiveObjectInspectorFactory.javaStringObjectInspector;
   }
 
-  protected String loadSource(SessionState sessionState, String functionName) throws UDFArgumentException {
-    Exec exec = sessionState.getDynamicVar(Exec.class);
-    try {
-      StoredProcedure storedProcedure = exec.getMsc().getStoredProcedure(
-              new StoredProcedureRequest(
-                      SessionState.get().getCurrentCatalog(),
-                      SessionState.get().getCurrentDatabase(),
-                      functionName));
-      return storedProcedure != null ? storedProcedure.getSource() : null;
-    } catch (TException e) {
-      throw new UDFArgumentException(e);
-    }
-  }
-
-  protected String functionName(ObjectInspector argument) {
-    ConstantObjectInspector inspector = (ConstantObjectInspector) (argument);
-    String functionCall = inspector.getWritableConstantValue().toString();
-    return functionCall.split("\\(")[0].toUpperCase();
-  }
-
   /**
    * Execute UDF
    */
   @Override
   public Object evaluate(DeferredObject[] arguments) throws HiveException {
-    if (exec == null) {
-      exec = new Exec();
```

Comment:
I saw that initializing the `Exec` instance is moved to the constructor of `Udf`. Could you please elaborate this change?

Reply:
Yeah, it was mostly because it looks odd init-ing Exec in `evaluate` method. It was getting initialized only once anyway, so it's better to move it to the constructor of Udf.